docker/README.md

```diff
@@ -0,0 +1,263 @@
+# Unity Catalog Dockerised Environment
+
+This project provides a Dockerised environment for running Unity Catalog. It includes everything needed to set up and interact with the catalog for testing purposes.
+
+## Prerequisites
+
+* Docker installed on your system.
+
+## Usage
+
+### 1. Building the Image
+
+The `build_docker.sh` script is responsible for building the Docker image for Unity Catalog. Run it from the project directory:
+
+```bash
+./bin/build-uc-server-docker
+```
+
+This will create an image named `unitycatalog`.
+
+### 2. Running the Catalog
+
+The `run_docker.sh` script starts the Unity Catalog container. It also creates a network named `unitycatalog_network` for the container to use. Run it from the project directory:
+
+```bash
+./bin.start-uc-server-in-docker
+```
+
+> [!TIP]
+> You can run the docker container with different settings such as using a different network, volume, or ports.
+> The run script is only for demo and happy path.
+
+This will start the container and make it accessible on port `8081` within the `unitycatalog_network` and create a volume named `unitycatalog_volume`.
+
+### 3. Adding Custom Startup Code
+
+> [!NOTE]
+> This feature has been removed temportarly until we find out the best way to add customisations to the containerisation process.
+
+### 4. Testing the Catalog
+
+This project provides several examples using `curl` commands to interact with the Unity Catalog API and demonstrate basic functionalities.
+
+#### 4.1. Store the Catalog Endpoint
+
+The examples use a variable `unitycatalog_endpoint` to store the catalog's URL. Update this variable with the actual address based on your network configuration.
+
+```bash
+unitycatalog_endpoint="http://unitycatalog:8080/api/2.1/unity-catalog"
+```
+
+> [!CAUTION]
+> Make sure you run the above command in your bash session before proceeding to the examples.
+> Otherwise your examples will not run correctly.
+
+> [!IMPORTANT]
+> Because the default run script runs the script on a network called `unitycatalog_network`, 
+> if you want to test the catalog you need to make sure you application is running in the same network.
+> For instance, you will not be able to test the API from your machine directly using Postman,
+> wget or cUrl unless you run them with the catalog on the same network, or you change the catalog's
```

Comment:
how do we run the application on the same network? does it have to be a docker application?

Reply:
This is confusing I have to admit. It's related to how networking works in docker.
To run two apps in the same network they don't need to be both docker containers, however, you need to ensure there is a path between your docker network and the application.
This can be achieved in various ways, such as using your host machine as the network, which allows your machine to interact with docker containers. However this changes between windows, Linux and mac.